tests/loops/test_loop_progress_integration.py

```diff
@@ -1,22 +1,19 @@
 from pytorch_lightning import Trainer
+from tests.loops.test_loops import _collect_loop_progress
 
 
 def test_loop_progress_integration():
     trainer = Trainer()
-    fit_loop = trainer.fit_loop
-    # check identities inside the fit loop
-    assert fit_loop.progress.epoch is fit_loop.epoch_loop.progress
-    assert fit_loop.epoch_loop.progress.batch is fit_loop.epoch_loop.batch_loop.progress
-    assert fit_loop.epoch_loop.progress.optim is fit_loop.epoch_loop.batch_loop.optim_progress
-    assert fit_loop.epoch_loop.progress.val is fit_loop.epoch_loop.val_loop.progress
-    assert fit_loop.epoch_loop.val_loop.progress.epoch is fit_loop.epoch_loop.val_loop.epoch_loop.progress
-    # check identities inside the evaluation and predict loops
-    assert trainer.validate_loop.progress.epoch is trainer.validate_loop.epoch_loop.progress
-    assert trainer.test_loop.progress.epoch is trainer.test_loop.epoch_loop.progress
-    assert trainer.predict_loop.progress.epoch is trainer.predict_loop.epoch_loop.progress
     # check no progresses are shared
-    assert trainer.fit_loop.progress is not trainer.validate_loop.progress
     assert trainer.validate_loop.progress is not trainer.test_loop.progress
     assert trainer.test_loop.progress is not trainer.predict_loop.progress
     # check the validation progresses are not shared
     assert trainer.fit_loop.epoch_loop.val_loop.progress is not trainer.validate_loop.progress
+    expected = _collect_loop_progress(trainer.fit_loop)["epoch_loop"]["progress"]
+    assert expected == trainer.fit_loop.epoch_loop.progress
+    expected = _collect_loop_progress(trainer.fit_loop)["epoch_loop"]["batch_loop"]["progress"]
+    assert expected == trainer.fit_loop.epoch_loop.batch_loop.progress
+    expected = _collect_loop_progress(trainer.fit_loop)["epoch_loop"]["val_loop"]["progress"]
+    assert expected == trainer.fit_loop.epoch_loop.val_loop.progress
+    expected = _collect_loop_progress(trainer.fit_loop)["epoch_loop"]["val_loop"]["epoch_loop"]["progress"]
+    assert expected == trainer.fit_loop.epoch_loop.val_loop.epoch_loop.progress
```

Comment:
This should avoid calling `_collect_loop_progress(trainer.fit_loop)` every time.

Also, the purpose of this test was to check that the references are correct, which are not there after your changes, so the whole test could be removed.

Reply:
This is still testing the references are correct.

Reply:
And yes, let's call _collect_loop_progress only once.

Reply:
> This is still testing the references are correct.

No because before was using `is` but now it's using `==`

Reply:
Oh, we can replace by is. They should be same reference.

Reply:
@carmocca Done !